### DISCUSSION TITLE
Happy new year I guess?

### DISCUSSION THREAD

**Discussion by cyrilchandelier** [etsy/open-api#1183] [Q&A]
I guess this happens every year, but I noticed the following breaking changes for the "when made" field:
- "2020_2023" becomes "2020_2024"
- "2004_2009" becomes "2005_2009"
- "before_2004" becomes "before_2005"
- "2000_2003" becomes "2000_2004"

It would be nice to know of these changes in advance, is this documented anywhere?

**Comment by Gareth064** [comment]
Unfortunately you will need to put a reminder in your calendar to check the API spec to see if these changed at the end of each year. Etsy won't tell us.

**Comment by sowsinsk** [comment]
+1 for calendar reminders

**Comment by simonj** [comment]
Oh! Thanks for heads up

**Comment by showtroylove** [comment]
I mentioned this in another post. You should use the API reference type, namely a string, for this field. It appears to be for display purposes only. Internally, you can utilize the dates associated with the listing. As recommended in the other post, avoid using enums with the Etsy API if a string will suffice. Adopting a defensive programming approach is advisable.

**Comment by showtroylove** [comment]
> I guess this happens every year, but I noticed the following breaking changes for the "when made" field:
> 
> * "2020_2023" becomes "2020_2024"
> * "2004_2009" becomes "2005_2009"
> * "before_2004" becomes "before_2005"
> * "2000_2003" becomes "2000_2004"
> 
> It would be nice to know of these changes in advance, is this documented anywhere?

It's documented in the API reference.  It was safe to intuit this would be a moving target.
